sync/sync.go

```diff
@@ -197,7 +197,7 @@ func (s *Synchronizer) verifierTask(ctx context.Context, block *core.Block, stat
 				} else {
 					s.HighestBlockHeader = highestBlock.Header
 
-					isBehind := s.HighestBlockHeader.Number > block.Number
+					isBehind := s.HighestBlockHeader.Number > block.Number+uint64(runtime.GOMAXPROCS(0))
```

Comment:
For some machines that institutional users use buffer might be as big as 64-128 blocks since CPUs these days have a lot of hw threads. Should we put a cap on this?

something like?

```suggestion
					isBehind := s.HighestBlockHeader.Number > block.Number+min(16, uint64(runtime.GOMAXPROCS(0)))
```

Reply:
Same way, we can put a limit on max fetcherTask number as well.

Reply:
I think it makes sense to add a limit for when to start catch-up mode otherwise 64-128 core CPU won't try to catch unless it is 64-128 blocks behind. However, we should not link it to max fetchers and verifiers because when the sync process initially starts we will not be fully utilising the resources, as a machine with 64-128 cores would only use 16 threads (I have put them into separate commits we can drop the second one if we don't want to limit the fetchers and verifiers).